### PR TITLE
Add conda env to setup instructions 

### PR DESCRIPTION
**Pull request by acatlett8**
## PR Summary
#21611 
## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ X] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/devel/development_setup.rst [hunk 1]
```diff
@@ -28,22 +49,27 @@ and activated with one of the following::
...
-If you have the proper privileges, you can use ``git@`` instead of
-``https://``, which works through the ssh protocol and might be easier to use
-if you are using 2-factor authentication.
+    conda activate mpl-dev
+
+Finish the install by the following command::
+
+    pip install -e . 
+
+Whenever you plan to work on Matplotlib, remember to `conda activate mpl-dev`
```
Comment (QuLogic) [inline doc/devel/development_setup.rst:71]: Oops, should have said double backticks:
```suggestion
Whenever you plan to work on Matplotlib, remember to ``conda activate mpl-dev``
```

File: doc/devel/development_setup.rst [hunk 2]
```diff
@@ -4,12 +4,33 @@
...
+:ref:`set-up-fork` for more details)::
+
+    git clone https://github.com/matplotlib/matplotlib.git
+
+This will place the sources in a directory :file:`matplotlib` below your
+current working directory.
+
+If you have the proper privileges, you can use ``git@`` instead of
+``https://``, which works through the ssh protocol and might be easier to use
+if you are using 2-factor authentication.
```
Comment (tacaswell) [inline doc/devel/development_setup.rst:22]:
```suggestion
```

No one should be pushing directly to the upstream fork (even if they can) as we do all work through PRs from individual forks.

**Comment by tacaswell** [comment]
I am 👍🏻  on this modulo my comment.

File: doc/devel/development_setup.rst (hunk 2 above)
Comment (QuLogic) [reply doc/devel/development_setup.rst:22]: Let's put that change into #23588?

Comment (tacaswell) [reply doc/devel/development_setup.rst:22]: Sounds good to me.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
@melissawm I am going to apologize in advance for causing you rebase-of-prose issues.

**Comment by tacaswell** [comment]
Congratulations @acatlett8 on your first merged Matplotlib PR 🎉  Hopefully we will hear from you again!

**Pull request merged**